src/background/services/deduplicator.test.ts

```diff
@@ -0,0 +1,323 @@
+import { Deduplicator } from './deduplicator';
+import { Logger } from '@/shared/logger';
+
+describe('Deduplicator', () => {
+  const deduplicatorService: Deduplicator = new Deduplicator({
+    logger: {
+      debug: jest.fn(),
+    } as unknown as Logger,
+  });
+  let returnValueFn1: {
+    access_token: { access: { type: string; actions: string[] }[] };
+  };
+  let returnValueFn2: {
+    access_token: { access: { type: string; actions: string[] }[] };
+  };
+
+  beforeAll(async (): Promise<void> => {
+    jest.useFakeTimers();
+  });
+
+  beforeEach(() => {
+    jest.runAllTimers();
+    returnValueFn1 = {
+      access_token: {
+        access: [
+          {
+            type: 'incoming-payment',
+            actions: ['create'],
+          },
+        ],
+      },
+    };
+    returnValueFn2 = {
+      access_token: {
+        access: [
+          {
+            type: 'quote',
+            actions: ['create'],
+          },
+        ],
+      },
+    };
+  });
+
+  // utility function to create async functions for testing
+  const createAsyncFn = <T>({
+    returnValue,
+    timeout = 0,
+    shouldReject = false,
+    mockFnName = 'mockFn',
+  }: {
+    returnValue: T;
+    timeout?: number;
+    shouldReject?: boolean;
+    mockFnName?: string;
+  }) => {
+    const fn = jest.fn(
+      async (..._args: unknown[]) =>
+        new Promise((resolve, reject) => {
+          try {
+            if (shouldReject) {
+              reject(new Error('Test error'));
```

Comment:
the `return` statement is not needed, I think; because the try-catch block is already controlling the flow of the promise.
If this promise rejects, any code after the try-catch block will not be executed anyways

Reply:
It'll still get called, just the promise will finalize to rejection.

![image](https://github.com/user-attachments/assets/9ea79c8b-2a14-4d05-b7cc-977dc6b7d9da)